packages/contracts-bedrock/src/L2/L2CrossDomainMessenger.sol

```diff
@@ -66,47 +83,25 @@ contract L2CrossDomainMessenger is CrossDomainMessenger, ISemver {
         override
         returns (bool)
     {
-        // Note: We are assuming that if message vaildation is active in the SystemConfig
-        //      the l1MessageValidator enforces the constraint that `_from` on the Deposit
-        //      is the L1CrossDomainMessenger.
-        //
-        // We make sure we only do validation on L1 -> L2 messages, otherwise
-        // this is a replay and should always be allowed.
+        // Early exit in case this a replay transaction. NOTE: this assumes that L1 message validation
+        // contract enforces forced arbitrary execution deposits to go through the CrossDomainMessenger contracts.
         if (!_isOtherMessenger()) {
             return true;
         }
-
-        address l2MessageValidator = L1Block(Predeploys.L1_BLOCK_ATTRIBUTES).l2MessageValidator();
-        if (l2MessageValidator == address(0)) {
-            return true;
+        // Check that we have enough gas to perform the message validation and finish the `relayMessage` call.
```

Comment:
Why not just include this in the first `hasMinGas` check in the `CrossDomainMessenger`. This should still be accessable across domains right? Only suggesting this because it seems more correct to check all gas throughout the duration of the function initially instead of midway through since we know at minimum we have enough gas for everything except any logic in this function, but we have already performed logic that could surpass the gas limit

Reply:
Yeah we can do that! The main difference in that type of an implementation is that we call `_relayMessageValidator` before the the first `hasMinGas` check.